app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -1,20 +1,60 @@
 package com.jetbrains.handson.mpp.mobile
 
+import android.content.Intent
+import android.net.Uri
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.widget.ArrayAdapter
+import android.widget.Button
+import android.widget.Spinner
 import android.widget.TextView
 
 class MainActivity : AppCompatActivity(), ApplicationContract.View {
 
+    private lateinit var presenter: ApplicationPresenter
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        presenter = ApplicationPresenter()
 
-        val presenter = ApplicationPresenter()
         presenter.onViewTaken(this)
     }
 
-    override fun setLabel(text: String) {
-        findViewById<TextView>(R.id.main_text).text = text
+    override fun setStationSubmitButtonText(text: String) {
+        findViewById<Button>(R.id.station_submit_button).text = text
+    }
+
+    override fun setStationSubmitButtonHandler() {
```

Comment:
Could we change this to be a function called by onCreate, rather than an override function? I don't think we need the presenter to call this